frontend/taipy-gui/src/components/Taipy/TimeSelector.spec.tsx

```diff
@@ -0,0 +1,156 @@
+            onchange: () => {},
+            addEventListener: () => {},
+            removeEventListener: () => {},
+            addListener: () => {},
+            removeListener: () => {},
+            dispatchEvent: () => false,
+        }),
+    });
+});
+
+afterEach(() => {
+    // @ts-ignore
+    delete window.matchMedia;
+});
+
+const curDate = new Date();
+curDate.setHours(1, 1, 1, 1);
+const curDateStr = curDate.toISOString();
+
+const cleanText = (val: string) => val.replace(/\u200e|\u2066|\u2067|\u2068|\u2069/g, "");
+
+describe("TimeSelector component", () => {
+    it("renders", async () => {
+        const { getByTestId } = render(
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <TimeSelector time={curDateStr} />
+            </LocalizationProvider>
+        );
+        const elt = getByTestId("ClockIcon");
+        expect(elt.parentElement?.tagName).toBe("BUTTON");
+    });
+    it("displays picker as a clock", async () => {
+        render(
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <TimeSelector time={curDateStr} asClock={true}/>
+            </LocalizationProvider>
+        );
+        const input = document.querySelector("input");
+        expect(input).toBeInTheDocument();
+    });
+    it("displays the right info for string", async () => {
+        const { getByTestId } = render(
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <TimeSelector time={curDateStr} className="taipy-time" />
+            </LocalizationProvider>
+        );
+        const elt = getByTestId("ClockIcon");
+        expect(elt.parentElement?.parentElement?.parentElement?.parentElement).toHaveClass("taipy-time-picker");
+        expect(elt.parentElement?.parentElement?.parentElement?.parentElement?.parentElement).toHaveClass("taipy-time");
+    });
+    it("displays the default value", async () => {
+        render(
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <TimeSelector
+                    defaultTime="2001-01-01T01:01:01.001Z"
+                    time={undefined as unknown as string}
+                />
+            </LocalizationProvider>
+        );
+        const input = document.querySelector("input");
+        expect(input).toBeInTheDocument();
+        expect(cleanText(input?.value || "").toLocaleLowerCase()).toEqual("01:01 am");
+    });
+    it("displays the default value with format", async () => {
+        render(
+            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <TimeSelector
+                    defaultTime="2001-01-01T01:01:01.001Z"
+                    time={undefined as unknown as string}
+                    format="hh aa"
+                />
+            </LocalizationProvider>
+        );
+        const input = document.querySelector("input");
+        expect(input).toBeInTheDocument();
+        expect(cleanText(input?.value || "")).toEqual("01 AM");
```

Comment:
Not sure if it's hour or minute ;-)

Reply:
It's hour + meridiem

Reply:
Alright, will change this with different numbers for hour, mins, sec